image/image.go

```diff
@@ -213,19 +213,24 @@ func makeFetcher(sto Store, dlOpts *DownloadOptions, db *asserts.Database) *addi
 func installCloudConfig(gadgetDir string) error {
 	var err error
 
-	cloudDir := filepath.Join(dirs.GlobalRootDir, "/etc/cloud")
+	cloudDir := filepath.Join(dirs.GlobalRootDir, "/etc/cloud/")
 	if err := os.MkdirAll(cloudDir, 0755); err != nil {
 		return err
 	}
 
 	cloudConfig := filepath.Join(gadgetDir, "cloud.conf")
 	if osutil.FileExists(cloudConfig) {
-		dst := filepath.Join(cloudDir, "cloud.cfg")
+		cfgDir := filepath.Join(dirs.GlobalRootDir, "/etc/cloud/cloud.cfg.d")
+		if err := os.MkdirAll(cfgDir, 0755); err != nil {
+			return err
+		}
+		dst := filepath.Join(cfgDir, "10_gadget_snap_cloud.cfg")
```

Comment:
What does it mean to have both cloud-init.cfg and the gadget config together?  Can we get any awkward outcome by having two independent parties tweaking each in isolation and then summing them up?